integration/benchnet2/automate/bootstrap_automation.go

```
@@ -0,0 +1,196 @@
+		Consensus: consensusNodes, Execution: executionNodes, Verification: verificationNodes}
+}
+
+func nodeStruct(nodeType string, nodeTemplatePath string, imageTag string, nodesData map[string]Node, nodeConfig map[string]int) map[string]NodeDetails {
+	nodeTemplate := createTemplate(nodeTemplatePath)
+	var nodeMap = make(map[string]NodeDetails)
+
+	for i := 1; i <= nodeConfig[nodeType]; i++ {
+		name := fmt.Sprint(nodeType, i)
+		replacementData := ReplacementData{NodeID: nodesData[name].NodeID, ImageTag: imageTag}
+
+		nodeString := replaceTemplateData(nodeTemplate, replacementData)
+		nodeStruct := unmarshalToStruct(nodeString, &NodeDetails{}).(*NodeDetails)
+		nodeStruct.Image = replacementData.ImageTag
+		nodeStruct.NodeID = replacementData.NodeID
+
+		nodeMap[name] = *nodeStruct
+	}
+
+	return nodeMap
+}
+
+func textReader(path string) []byte {
+	file, err := os.ReadFile(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return file
+}
+
+func writeYamlBytesToFile(filepath string, yamlData []byte) {
+	file, err := os.OpenFile(filepath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	_, err = file.Write(yamlData)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	file.Close()
+}
+
+func replaceTemplateData(template *template.Template, data ReplacementData) string {
+	var doc bytes.Buffer
+	err := template.Execute(&doc, data)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return doc.String()
+}
+
+func createTemplate(templatePath string) *template.Template {
+	template, err := template.New("todos").Parse(string(textReader(templatePath)))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return template
+}
+
+func unmarshalToStruct(source string, target interface{}) interface{} {
+	err := yaml.Unmarshal([]byte(source), target)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return target
+}
+
+func marshalToYaml(source interface{}, outputFilePath string) {
+	yamlData, err := yaml.Marshal(source)
+	if err != nil {
+		log.Fatal(err)
+	}
```

We need to return the error here after logging it.